File: leetcode/go/0401-binaryWatch.go
//go:build ignore

package main

import (
	"fmt"
)

func countBits(n int) int {
	cnt := 0
	for ; n > 0; n >>= 1 {
		cnt += n & 1
	}
	return cnt
}

func readBinaryWatch(turnedOn int) []string {
	res := []string{}

	for h := 0; h < 12; h++ {
		for m := 0; m < 60; m++ {
			if countBits(h)+countBits(m) == turnedOn {
				timeStr := fmt.Sprintf("%d:%02d", h, m)
				res = append(res, timeStr)
			}
		}
	}

	return res
}

func main() {
	turnedOn := 1
	fmt.Println(readBinaryWatch(turnedOn))
}

File: leetcode/python/0401-binaryWatch.py
from typing import List

class Solution:
    def countBits(self, n: int) -> int:
        cnt = 0
        while n:
            cnt += n & 1
            n >>= 1
        return cnt
    
    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        res = []
        for h in range(12):
            for m in range(60):
                if self.countBits(h) + self.countBits(m) == turnedOn:
                    res.append(f"{h}:{m:02d}")
        return res
    
def main():
    s = Solution()
    print(s.readBinaryWatch(1))

if __name__ == "__main__":
    main()
    